browser/src/control/Control.SheetsBar.js

```diff
@@ -104,6 +123,86 @@ class SheetsBar {
 		}
 	}
 
+	showSheetListPopup() {
+		const buttonElement = document.getElementById('sheets-buttons-toolbox');
+
+		if (!this.partNames || this.partNames.length === 0) {
+			return;
+		}
+
+		let sheetEntries = [];
+		
+		for (let i = 0; i < this.partNames.length; i++) {
+			if (app.calc.isPartHidden(i)) {
+				continue;
+			}
+
+			let displayText = this.partNames[i];
+			sheetEntries.push({
+				columns: [ { text: displayText } ],
+				row: i,
+				selected: i === this.selectedPart
+			});
+		}
+
+		let json = {
+			id: 'SheetListPopup-dropdown',
+			type: 'dropdown',
+			jsontype: 'dialog',
+			cancellable: true,
+			popupParent: buttonElement,
+			clickToClose: '_POPOVER_',
+			children: [
+				{
+					id: 'container-dropdown',
+					type: 'container',
+					enabled: true,
+					vertical: true,
+					children: [
+						{
+							type: 'treelistbox',
+							id: 'sheetnames',
+							entries: sheetEntries
+						}
+					]
+				}
+			]
+		};
+
+		let popupBuildEvent = {
+			data: json,
+			callback: this.onSelectSheet.bind(this),
+		};
+
+		this.map.fire('jsdialog', popupBuildEvent);
+
+		return json;
+	}
+
+	onSelectSheet(element, action, data, index) {
```

Comment:
This has to be moved to a custom callback for JSDialog so it will be executed when submenu entry is used.
See my comment next to builder in this file